### PR TITLE
Don't decompose addmm, take 2 

### PR DESCRIPTION
**Pull request by ngimel**
I messed up #912 with merge gone wrong, this is take 2, conditionally decomposes if we don't codegen addmm with triton. Perf numbers in #912 still hold.

### CODE REVIEW

File: torchinductor/decomposition.py [hunk 1]
```diff
@@ -106,16 +106,19 @@ def clamp(x, min=None, max=None):
...
-def addmm(input, mat1, mat2):
-    return torch.mm(mat1, mat2) + input
-
-
 @register_decomposition([aten.tanh])
 def tanh(x):
     return 2.0 / (1.0 + torch.exp(-2.0 * x)) - 1.0
 
 
+@register_decomposition([aten.addmm])
```
Comment (Chillee) [inline torchinductor/decomposition.py:114]: Perhaps we should put these in a special section.

**Review by Chillee — APPROVED** [review]
Do we need to make a new `ExterneKernelOut` class everytime we want to use an out variant?

**Comment by ngimel** [comment]
> Do we need to make a new `ExterneKernelOut` class everytime we want to use an out variant?

I don't know tbh, was copying what was done in other kernels

File: torchinductor/decomposition.py (hunk 1 above)
Comment (ngimel) [reply torchinductor/decomposition.py:114]: I don't expect many of conditional decomps, if there are a few we can combine them.

**Comment by jansel** [comment]
> Do we need to make a new `ExterneKernelOut` class everytime we want to use an out variant?

The automatic fallbacks burn in shapes, so for stuff we want to be long lived we should define shape formulas for them.

**Comment by Chillee** [comment]
@jansel Can we just re-use core's shape propagation formulas?

**Comment by jansel** [comment]
> @jansel Can we just re-use core's shape propagation formulas?

Yes, that sounds like a great idea.  Is it possible to do that now?

**Comment by ngimel** [comment]
Test error is oom, we should reduce batch sizes.

**Pull request merged**